api/src/routes/user.test.ts

```diff
@@ -87,6 +93,88 @@ describe('userRoutes', () => {
         expect(user).toMatchObject(baseProgressData);
       });
     });
+    describe('/user/user-token', () => {
+      beforeEach(async () => {
+        const user = await fastifyTestInstance.prisma.user.findFirst({
+          where: { email: 'foo@bar.com' }
+        });
+
+        await fastifyTestInstance.prisma.userToken.deleteMany({
+          where: {
+            userId: user?.id
+          }
+        });
+      });
+
+      // TODO(Post-MVP): consider using PUT and updating the logic to upsert
+      test('POST success response includes a JWT encoded string', async () => {
+        const response = await superRequest('/user/user-token', {
+          method: 'POST',
+          setCookies
+        });
+
+        const userToken = response.body.userToken;
+        const decodedToken = jwt.decode(userToken);
+
+        expect(response.body).toStrictEqual({ userToken: expect.any(String) });
+        expect(decodedToken).toStrictEqual({
+          userToken: expect.stringMatching(/^[\w0-9]{64}$/),
```

Comment:
CodeQL beat me to it

```suggestion
          userToken: expect.stringMatching(/\w{64}$/),
```

Reply:
Otherwise

```suggestion
          userToken: expect.stringMatching(/^[a-zA-Z0-9]{64}$/),
```

Reply:
Thanks, I completely forgot about _. To be honest, it's just a guess about what loopback does and I can't see a reason why it would matter.

Still, let's be precise and use your fix.